src/openapi/streaming/streaming.ts

```diff
@@ -492,12 +517,12 @@ class Streaming extends MicroEmitter<EmittedEvents> {
         }
 
         if (!Array.isArray(updates)) {
-            this.processUpdate(updates);
-            return;
+            updates = [updates];
         }
 
-        for (let i = 0; i < updates.length; i++) {
-            const update = updates[i];
+        for (const update of updates) {
+            this.contextMessageCount++;
+            this.contextLastMessageReceived = new Date().toISOString();

```

Comment:
If you rebase from master I've added a lastActivity already for this- can you delete this and use my field?
its just a number but I think thats good to avoid us doing this for every streaming message.